### PR TITLE
cleanup: replace dial with newclient

### PR DESCRIPTION
**Pull request by janardhanvissa**
Partially address: https://github.com/grpc/grpc-go/issues/7049

RELEASE NOTES: None

### CODE REVIEW

File: internal/idle/idle_e2e_test.go [hunk 1]
```diff
@@ -131,20 +131,24 @@ func (s) TestChannelIdleness_Disabled_NoActivity(t *testing.T) {
...
-	r.UpdateState(resolver.State{Addresses: []resolver.Address{{Addr: backend.Address}}})
+	r.InitialState(resolver.State{Addresses: []resolver.Address{{Addr: backend.Address}}})
 
-	// Verify that the ClientConn moves to READY.
+	// Trigger the resolver by initiating an RPC.
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
 	defer cancel()
+	go func() {
+		_ = cc.Invoke(ctx, "/test/method", nil, nil)
+	}()
```
[the remark this answers is not in the document]
Comment (janardhanvissa) [reply internal/idle/idle_e2e_test.go]: Done

File: orca/producer_test.go [hunk 2]
```diff
@@ -153,12 +153,12 @@ func (s) TestProducer(t *testing.T) {
 	li := &listenerInfo{scChan: make(chan balancer.SubConn, 1), listener: oobLis, opts: lisOpts}
 	addr := setListenerInfo(resolver.Address{Addr: lis.Addr().String()}, li)
 	r.InitialState(resolver.State{Addresses: []resolver.Address{addr}})
-	cc, err := grpc.Dial("whatever:///whatever", grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"customLB":{}}]}`), grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	cc, err := grpc.NewClient("whatever:///whatever", grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"customLB":{}}]}`), grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
```
[the remark this answers is not in the document]
Comment (janardhanvissa) [reply orca/producer_test.go]: Done

File: orca/producer_test.go [hunk 3]
```diff
@@ -319,10 +319,11 @@ func (s) TestProducerBackoff(t *testing.T) {
 	lisOpts := orca.OOBListenerOptions{ReportInterval: reportInterval}
 	li := &listenerInfo{scChan: make(chan balancer.SubConn, 1), listener: oobLis, opts: lisOpts}
 	r.InitialState(resolver.State{Addresses: []resolver.Address{setListenerInfo(resolver.Address{Addr: lis.Addr().String()}, li)}})
-	cc, err := grpc.Dial("whatever:///whatever", grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"customLB":{}}]}`), grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	cc, err := grpc.NewClient("whatever:///whatever", grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"customLB":{}}]}`), grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
```
[the remark this answers is not in the document]
Comment (janardhanvissa) [reply orca/producer_test.go]: Done

File: resolver/manual/manual_test.go [hunk 4]
```diff
@@ -59,15 +59,17 @@ func TestResolver(t *testing.T) {
 	})
 
 	t.Run("happy_path", func(t *testing.T) {
-		_, err := grpc.Dial("whatever://localhost",
+		r.InitialState(resolver.State{Addresses: []resolver.Address{
+			{Addr: "ok"},
+		}})
```
[the remark this answers is not in the document]
Comment (janardhanvissa) [reply resolver/manual/manual_test.go]: Done

File: test/authority_test.go [hunk 5]
```diff
@@ -194,11 +194,14 @@ func (s) TestColonPortAuthority(t *testing.T) {
...
 	// Append "localhost" before calling net.Dial, in case net.Dial on certain
 	// platforms doesn't work well for address without the IP.
-	cc, err := grpc.Dial(":"+port, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
-		return (&net.Dialer{}).DialContext(ctx, "tcp", "localhost"+addr)
+	cc, err := grpc.NewClient(":"+port, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+		if len(addr) > 0 && addr[0] == ':' {
+			addr = "localhost" + addr
+		}
+		return (&net.Dialer{}).DialContext(ctx, "tcp", addr)
 	}))
```
[the remark this answers is not in the document]
Comment (janardhanvissa) [reply test/authority_test.go]: Done

**Review by arjan-bal — APPROVED** [review]
LGTM!

File: test/authority_test.go [hunk 6]
```diff
@@ -194,11 +194,9 @@ func (s) TestColonPortAuthority(t *testing.T) {
 	//
 	// Append "localhost" before calling net.Dial, in case net.Dial on certain
 	// platforms doesn't work well for address without the IP.
```
Comment (arjan-bal) [inline test/authority_test.go:196]: Please remove this comment as it's not longer applicable.

**Review by dfawley — APPROVED** [review]

**Pull request merged**